docs/processing-performance.asciidoc

```diff
@@ -5,40 +5,110 @@ APM Server performance depends on a number of factors: memory and CPU available,
+
+// Results
+You can use the results of our tests to guide your sizing decisions, however, performance will vary based on factors unique to your use case like your specific setup, the size of APM event data, and the exact number of agents.
+
+:hardbreaks-option:
+
 [options="header"]
-|=======================================================================
-|Transaction/Instance |512 MB Instance |2 GB Instance |8 GB Instance
-|Small transactions
+|====
+| Profile / Cloud | AWS | Azure | GCP (Agent) | GCP (OTel)
+
+| *1 GB*
+(10 agents)
+| 9,600
+events/second
+| 6,400
+events/second
+| 9,600
+events/second
+| 28,000
+events/second
+
+| *4 GB*
+(30 agents)
+| 25,500
+events/second
+| 18,000
+events/second
+| 17,800
+events/second
+| 49,300
+events/second
+ 
+| *8 GB*
+(60 agents)
+| 40,500
+events/second
+| 26,000
+events/second
+| 25,700
+events/second
+| 89,300
+events/second
 
-_5 spans with 5 stack frames each_ |600 events/second |1200 events/second |4800 events/second
-|Medium transactions
+| *16 GB*
+(120 agents)
+| 72,000
+events/second
+| 51,000
+events/second
+| 45,300
+events/second
+| 145,000
+events/second
 
-_15 spans with 15 stack frames each_ |300 events/second |600 events/second |2400 events/second
-|Large transactions
+| *32 GB*
+(240 agents)
+| 135,000
+events/second
+| 95,000
+events/second
+| 95,000
+events/second
+| 381,000
+events/second
 
-_30 spans with 30 stack frames each_ |150 events/second |300 events/second |1400 events/second
-|=======================================================================
+|====
 
-In other words, a 512 MB instance can process \~3 MB per second,
-while an 8 GB instance can process ~20 MB per second.
+:!hardbreaks-option:
 
-APM Server is CPU bound, so it scales better from 2 GB to 8 GB than it does from 512 MB to 2 GB.
-This is because larger instance types in {ecloud} come with much more computing power.
+// In other words, a 512 MB instance can process \~3 MB per second,
+// while an 8 GB instance can process ~20 MB per second.
+
+APM Server is CPU bound and larger instance types in {ecloud} come with much more computing power.
+// so it scales better from 2 GB to 8 GB than it does from 512 MB to 2 GB.
 
 Don't forget that the APM Server is stateless.
 Several instances running do not need to know about each other.
 This means that with a properly sized {es} instance, APM Server scales out linearly.
 
-NOTE: RUM deserves special consideration. The RUM agent runs in browsers, and there can be many thousands reporting to an APM Server with very variable network latency.
\ No newline at end of file
+NOTE: RUM deserves special consideration. The RUM agent runs in browsers, and there can be many thousands reporting to an APM Server with very variable network latency.
+
+// [discrete]
+// ==== Troubleshoot sizing issues
+
+// [discrete]
+// ===== Scale the APM Server and ES
+
+// [discrete]
+// ===== Apply head based sampling
+
+// [discrete]
+// ===== Apply tail based sampling
```

Comment:
@simitt in https://github.com/elastic/apm-server/issues/7840 you mentioned adding information on troubleshooting sizing issues. Should we add a section like this? Is there any source content to help me build out this section?

Reply:
For scaling ES, we have docs [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/scalability.html). I'll leave scaling APM Server to @simitt.

You should be able to pull information for the sampling section from [these docs](https://www.elastic.co/guide/en/apm/guide/current/sampling.html). The tl;dr is that sampling aims to lower the amount of data ingested (and the effort required to analyze that data).

@simitt what about adding:
* span compression?
* stack trace collection?

@colleenmcginnis if Silvia gives that a thumbs up on the two topics above, the docs for span compression are [here](https://www.elastic.co/guide/en/apm/guide/current/span-compression.html) and we have some info on stack trace info [here](https://www.elastic.co/guide/en/apm/guide/current/reduce-apm-storage.html#reduce-stacktrace).

Reply:
@elastic/apm-server can anyone take a crack at these questions in Silvia and Andrew's absence?

Reply:
> span compression?
stack trace collection?

Both of these will have some impact on APM-Server performance but I think the impact will be more on storage (ES) and agent end. That being said, adding these sounds good to me.

Reply:
Recently I spent some time studying the requirements of TBS. See https://github.com/elastic/apm-server/issues/11544
Other than the explicit disk requirement, there's an implicit requirement on memory for TBS to function efficiently. https://github.com/elastic/apm-server/pull/11662 added a rather vague statement about that.

On the same topic we still have follow up issues https://github.com/elastic/apm-server/issues/11346 and https://github.com/elastic/apm-dev/issues/1064 . But I'd say we don't any concrete numbers yet.

Reply:
Recently I spent some time studying the requirements of TBS. See #11544
Other than the explicit disk requirement, there's an implicit requirement on memory for TBS to function efficiently. #11662 added a rather vague statement about that.

On the same topic we still have follow up issues #11346 . But I'd say we don't any concrete numbers yet.

Reply:
I suggest we do not go into any details here, but rather add a sentence/paragraph along the lines of, in case your apm-server cannot be scaled to the amount of load that you are expecting, check out the section around [reducing storage](https://www.elastic.co/guide/en/apm/guide/current/reduce-apm-storage.html) for options to decrease the ingestion volume.